### ISSUE TITLE
Update broken and Gyro (rotation) is way to nervous

### ISSUE THREAD

**Issue by ghost** [koreader/koreader#12674]
* KOReader version: [42d3f25](https://github.com/koreader/koreader/commit/42d3f25034778c121dc458b3d5246896fedc5db2)
 
* Device: Kobo Sage

#### Issue
I have just updated koreader. The old version was something  2024.7-100.

First of all the update download without errors. The installation step did not succeed -took a few minutes. Putting the device to sleep showed the current book cover, waking the device up did nothing.

A hard reset and a consecutive koreader start triggered the update (fbink messages ....)

Now the device works almost as expected.

## BUT

The device is configured to take gyro events for 180 degree turns.

If the device is horizontally on a shaky table, any tiny shake of the table (pushing with my legs) turns the screen by 180°.
That is a new behavior and unwanted.

Sorry I can not provide a crash.log by now.

**Comment by Frenzie** [comment]
Pinging @mergen3107 since I don't recall the details of what kind of stuff has changed.

**Comment by mergen3107** [comment]
@Frenzie 
This is about Kobo (you put Kindle label), I have no clue

**Comment by NiLuJe** [comment]
I don't think anything should really have changed on that front.

i.e., if we rotate, it's because we got a rotate event, and whether we get a rotate event or not is entirely out of our hands ;o).

**Comment by Frenzie** [comment]
That's the reason for my mistake: Kindle is the only thing that may have changed afaik.

**Comment by NiLuJe** [comment]
I mean, there might have been a Nickel update that changed the default Z-tilt setting on that device, but, again, that's out of our hands ;).

FWIW, just updated a Sage from 2024.04-something to current, and it looks fine.

I don't use the gyro, so I can't vouch for however it's supposed to behave, but it rotates sanely, albeit relatively early if you just tilt it (e.g., something like 35° or so).

**Comment by NiLuJe** [comment]
For reference, in the event0 input stream,

`EV_MSC:MSC_RAW:24` is UR and 23 is UD
And 28 is FU (face up) and 27 is FD (face down)

I'd take particular notice of whether the Face events look sane (we don't use them, but they should give a good indication of whether the Z detection isn't going wonky).

**Comment by NiLuJe** [comment]
> I mean, there might have been a Nickel update that changed the default Z-tilt setting on that device, but, again, that's out of our hands ;).

c.f.,

`/sys/bus/i2c/drivers/kx122/1-001e/input/input1/event1/tilt_angle_ll` & `/sys/bus/i2c/drivers/kx122/1-001e/input/input1/event1/tilt_angle_hl`

**Comment by NiLuJe** [comment]
> IIRC, there's a minimal UI about it the the Nickel Dev menu, though.

Yep, Nickel says 14° for the Z-lock & 80° for the Back/Front flip.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
